core/src/main/java/hudson/search/Search.java

```diff
@@ -76,7 +76,7 @@ public void doIndex(StaplerRequest req, StaplerResponse rsp) throws IOException,
                     SuggestedItem target = find(index, query, smo);
                     if(target!=null) {
                         // found
-                        rsp.sendRedirect2(a.getUrl()+target.getUrl());
+                        rsp.sendRedirect2(req.getContextPath()+target.getUrl());
```

Comment:
Right. But that works even if I don't apply my changes because `goTo` requests an HTML page within Jenkins dashboard context and I need to do it from a different context `"job/folder/`. I am trying to figure out how to do this.